src/librustc/mir/mod.rs

```diff
@@ -1773,7 +1773,7 @@ rustc_index::newtype_index! {
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct PlaceRef<'a, 'tcx> {
-    pub local: &'a Local,
+    pub local: Local,
     pub projection: &'a [PlaceElem<'tcx>],
```

Comment:
Orthogonal, but the 'a here could be 'tcx, right?

Reply:
Unsure, I just did it to try out and I'm getting ...

```
error[E0597]: `deref` does not live long enough
    --> src/librustc_mir/borrow_check/mod.rs:1400:41
     |
855  |   impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
     |             ---- lifetime `'tcx` defined here
...
1400 |                   root_place.projection = &deref;
     |                                           ^^^^^^ borrowed value does not live long enough
...
1413 |           if places_conflict::borrow_conflicts_with_place(
     |  ____________-
1414 | |             self.infcx.tcx,
1415 | |             &self.body,
1416 | |             place,
...    |
1420 | |             places_conflict::PlaceConflictBias::Overlap,
1421 | |         ) {
     | |_________- argument requires that `deref` is borrowed for `'tcx`
...
1433 |       }
     |       - `deref` dropped here while still borrowed

error[E0621]: explicit lifetime required in the type of `issued_borrow`
   --> src/librustc_mir/borrow_check/diagnostics/conflict_errors.rs:547:9
    |
344 |         issued_borrow: &BorrowData<'tcx>,
    |                        ----------------- help: add explicit lifetime `'cx` to the type of `issued_borrow`: `&'cx borrow_check::borrow_set::BorrowData<'tcx>`
...
547 |         err
    |         ^^^ lifetime `'cx` required

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0597, E0621.
For more information about an error, try `rustc --explain E0597`.
error: could not compile `rustc_mir`.

To learn more, run the command again with --verbose.
```

Didn't pay a lot of attention to be honest, should I fix this in this PR and investigate on a different one?.

Reply:
no let's do it in another PR.

Reply:
I think you should start by replacing `PlaceRef<'_, 'tcx>` with `PlaceRef<'tcx, 'tcx>` and see where that breaks down (most things should be able to handle it, although... unsure how useful it is?).

Reply:
This was addressed here https://github.com/rust-lang/rust/pull/69714